src/Provider/AbstractProvider.php

```diff
@@ -216,6 +218,54 @@ public function getAccessToken($grant = 'authorization_code', $params = [])
         return $grant->handleResponse($result);
     }
 
+    /**
+     * Get an authenticated request instance.
+     *
+     * Creates a PSR-7 compatible request instance that can be modified.
+     * Often used to create calls against an API that requires authentication.
+     *
+     * @param  string $method
+     * @param  string $url
+     * @param  AccessToken $token
+     * @return RequestInterface
+     */
+    public function getAuthenticatedRequest($method, $url, AccessToken $token)
+    {
+        $request = new Request($url, $method);
```

Comment:
Should this call `getHttpClient()` to build the request, so that the user may choose whatever HTTP client they wish to use?